### ISSUE TITLE
Static singletons used everywhere

### ISSUE THREAD

**Issue by stevebauman** [facebook/facebook-php-business-sdk#604]
I'm working on an application that will facilitate reporting against multiple Meta integrations during the same request, but I quickly discovered that this is will cause issues if you keep one integration in memory and call it later after another one has been initialized. For example:

```php
use FacebookAds\Api;
$apiOne = Api::init(...);

$apiTwo = Api::init(...); // Overwrites the first

$apiOne->... // Method calls will now occur on 
```

**Comment by norberttech** [comment]
I'm facing similar problem... 

@stevebauman did you managed to find a solution for that?

**Comment by stevebauman** [comment]
@norberttech No I haven't unfortunately. 

I'm considering just using raw HTTP calls to their API instead. Doesn't seem like there's an easy way with this SDK without extending a bunch of classes and overriding a ton of methods.

**Comment by norberttech** [comment]
yeah, I was thinking the same but it also seems that [joelbutcher/facebook-graph-sdk](https://packagist.org/packages/joelbutcher/facebook-graph-sdk) works fine, I just tested fetching campaigns and it works, plus it's not a singleton so multitenancy is doable. 

```php
$fb = new \Facebook\Facebook([
  'app_id' => $authConfig['app_id'],
  'app_secret' => $authConfig['app_secret'],
  'default_graph_version' => $authConfig['api_version'],
  'default_access_token' => 'long_living_access_token',
  'http_client_handler' => new GuzzleHttpClient(...), // you can also inject your custom client in case you would like to mock responses
]);
            
$accountsData = $facebook->get('/me/adaccounts')->getDecodedBody();
$campaignsData = $facebook->get($accountsData['data'][0]['id'] . '/campaigns')->getDecodedBody();
```

> I'm considering just using raw HTTP calls to their API instead. 

Yeah, in case of any issues with this sdk I'm probably also gonna look into simple http request/response that would return associative arrays.
